core/src/Result.cpp

```diff
@@ -30,8 +30,8 @@ Result::Result(std::wstring&& text, ByteArray&& rawBytes, std::vector<ResultPoin
 
 Result::Result(DecoderResult&& decodeResult, std::vector<ResultPoint>&& resultPoints, BarcodeFormat format)
     : _status(decodeResult.errorCode()),
-      _text(std::move(decodeResult).text()),
-      _rawBytes(std::move(decodeResult).rawBytes()),
+      _text(std::move(decodeResult.text())),
+      _rawBytes(std::move(decodeResult.rawBytes())),
```

Comment:
Note also this unnoticeable issue/bug that I saw just by chance.

Reply:
There was actually no issue before and now you introduced one: your change introduces a string/array copy where none was present before. Note the rvalue reference getters in DecoderResult used to steal the contests of `text/rawBytes`. For them to be selected, the object they are called on need to be an rvalue reference. Since `decodeResult` has a 'name' in this context, it is an lvalue reference and needs to be cast back with `std::move`. You are calling `std::move` on a `const &`, which makes no sense and will result in calling the copy constructor.

Reply:
You are totally right, there were no copies before, and my changes create copies. My confusion came from my assumption (without checking, my bad) that text() and rawBytes() return `const &` and `&` for const and non-const object respectively, but in fact, they return  `const &` and `&&`.